### ISSUE TITLE
to_timezone() outputs wrong time zone designator (or wrong timestamp?)

### ISSUE THREAD

**Issue by salsasepp** [questdb/questdb#2364]
### Describe the bug

This UNIX timestamp (with microseconds)
```
1659078123000000
```
corresponds to this ISO datetime:
```
2022-07-29T07:02:03.000000Z
```
When converting to a different time zone using to_timezone(), the timestamp string is "almost" correct, but has a Z timezone designator, which is wrong.

### To reproduce

```
SELECT to_timezone(1659078123000000, 'Europe/Berlin');
```
outputs
```
2022-07-29T09:02:03.000000Z
```

### Expected Behavior

Either a local time (no time zone designator)
```
2022-07-29T09:02:03.000000
```
or a timestamp with UTC offset
```
2022-07-29T09:02:03.000000+02:00
```
Alternatively, in case to_timezone() is expected to output UTC (which I would find confusing, given the description of the function):
```
2022-07-29T07:02:03.000000Z
```

### Environment

```markdown
- **QuestDB version**: 6.4.3 (binaries)
- **OS**: Linux desktop 5.15.0-43-generic #46-Ubuntu SMP Tue Jul 12 10:30:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by bluestreak01** [comment]
Perhaps we can return make return value of `to_timezone()` a `string` ?

**Comment by salsasepp** [comment]
Making `to_timezone()` return a ISO8601 string seems a good idea to me.

For my use case, dealing with time zones is purely a business logic convenience (displaying a date/time in the local time zone of a stock exchange), I actually prefer to have "clean UTC" in the database and mess with time zones strictly in application code only. However, if we need a `to_timezone()` function at all, I believe having a correct and confusion-free ISO string is a good thing.

**Comment by zdytch** [comment]
> QuestDB doesn't have timestamp_with_timezone datatype, so that to_timezone() return value is interpreted as TIMESTAMP (always UTC). While I fully agree this is confusing, what are the practical implications of this? For example this will produce the desired output:
> 
> SELECT to_str(to_timezone(1659078123000001, 'Europe/Berlin'), 'y-MM-ddTHH:mm:ss.SSSUUU+02:00');

You can’t hardcode +02:00, as the offset changes with daylight saving time—it becomes +03:00 in summer.

**Comment by zdytch** [comment]
> Making `to_timezone()` return a ISO8601 string seems a good idea to me.
> 
> For my use case, dealing with time zones is purely a business logic convenience (displaying a date/time in the local time zone of a stock exchange), I actually prefer to have "clean UTC" in the database and mess with time zones strictly in application code only. However, if we need a `to_timezone()` function at all, I believe having a correct and confusion-free ISO string is a good thing.

I agree. The presence of the to_timezone() function suggests that the timezone must accompany its corresponding value, as the timezone forms an essential part of the context. Without it, the value lacks meaning

**Comment by nwoolmer** [comment]
We are changing how the timestamp type interacts with PG Wire at the moment.

The 'Z' suffix can be confusing, but is a UI symptom only. You can retrieve an epoch timestamp directly by using a conversion to long.

However, it would be nice to support an offset timestamp format in our printed output, as we do for alternative logging timezones.
